mono/mini/mini.h

```diff
@@ -787,6 +787,11 @@ struct MonoCallInst {
 	guint tailcall : 1;
 	/* If this is TRUE, 'fptr' points to a MonoJumpInfo instead of an address. */
 	guint fptr_is_patch : 1;
+#if __cplusplus
+	MonoJitICallId jit_icall_id : 9; // 9 is checked in jit-icall-reg.h mono_find_jit_icall_info
```

Comment:
Why is the ifdef needed ?

Reply:
Not standard C, though, I had forgotten, that msvc supports it, along with gcc/clang.
So we can just use the C++ form until/unless another C compiler matters (I expect that'd be a large battle more generally, but I try to remain open to it).

Reply:
Remove `#if` and depend on msvc/gcc/clang extension -- "bitfield type other than int"?
Gcc has stuff like this -- enum bitfields, subject to autoconf, to work with non-gcc macros.

Other choice is make the bitfield-ness conditional, not the enum-ness:
```
#if __cplusplus || __GNUC__ || __clang__ || _MSC_VER
 enum foo f : 9
#else 
 enum foo f;
```

Could also just not use bitfield, arguably call instructions are rare and can be larger, and the IR is short-lived -- method at a time not file at a time, right?

Reply:
Just make it a byte.

Reply:
It needs 9 bits.

Do you mean, just make it an enum, 
not bitfield,
 don’t worry about the size?

It’ll be pointer-sized in C++, 32bits in C.
Pointer-sized because widened for big endian.

I’ll do that, on the assumption it is what you mean, not to disagree.
The #else case above after gcc/clang/mscver.

Another choice is short or int, and likely cast in C++.

Reply:
Hopefully ok.
```
Author: Jay Krell <jaykrell@microsoft.com>
Date:   Mon Jun 10 12:33:45 2019 -0700
    PR: Don't worry about size of MonoCallInst.jit_icall_id. Let it be 32 or 64 bits instead of 9.
--- a/mono/metadata/jit-icall-reg.h
+++ b/mono/metadata/jit-icall-reg.h
@@ -389,7 +389,7 @@ mono_find_jit_icall_info (MonoJitICallId id)
-       g_static_assert (MONO_JIT_ICALL_count < 0x200); // fits in 9 bits, see MonoCallInst
+       g_static_assert (MONO_JIT_ICALL_count < 0x200); // fits in 9 bits
--- a/mono/mini/mini.h
+++ b/mono/mini/mini.h
@@ -787,11 +787,7 @@ struct MonoCallInst {
-#if __cplusplus
-       MonoJitICallId jit_icall_id : 9; // 9 is checked in jit-icall-reg.h mono_find_jit_icall_info
-#else
-       guint jit_icall_id : 9; // 9 is checked in jit-icall-reg.h mono_find_jit_icall_info
-#endif
+       MonoJitICallId jit_icall_id;
```